covquestions-editor-app/src/components/questionnaireEditor/QuestionnaireFormEditor.tsx

```diff
@@ -156,8 +155,23 @@ export function QuestionnaireFormEditor(props: QuestionnaireFormEditorProps) {
     }
   };
 
+  const handleAddItem = (section?: SectionTypeArray) => {
+    if (!isNonArraySection(activeItem.section)) {
+      dispatch(
+        addNewItem({
+          section: section || activeItem.section,
+          template:
+            section === undefined
+              ? questionnaireInEditor.questionnaire[activeItem.section]![activeItem.index]
```

Comment:
This could be a little more typesafe as: 
questionnaireInEditor.questionnaire[activeItem.section]?.[activeItem.index]